src/platform/linux/inputtino.cpp

```diff
@@ -0,0 +1,648 @@
+    if (std::holds_alternative<inputtino::PS5Joypad>(*gamepad->joypad)) {
+      if (touch.pressure > 0.5) {
+        std::get<inputtino::PS5Joypad>(*gamepad->joypad).place_finger(touch.pointerId, touch.x * inputtino::PS5Joypad::touchpad_width, touch.y * inputtino::PS5Joypad::touchpad_height);
+      }
+      else {
+        std::get<inputtino::PS5Joypad>(*gamepad->joypad).release_finger(touch.pointerId);
+      }
+    }
+  }
+
+  void
+  gamepad_motion(input_t &input, const gamepad_motion_t &motion) {
+    auto raw = (input_raw_t *) input.get();
+    auto gamepad = raw->gamepads[motion.id.globalIndex];
+    if (!gamepad) {
+      return;
+    }
+    // Only the PS5 controller supports motion
+    if (std::holds_alternative<inputtino::PS5Joypad>(*gamepad->joypad)) {
+      switch (motion.motionType) {
+        case LI_MOTION_TYPE_ACCEL:
+          std::get<inputtino::PS5Joypad>(*gamepad->joypad).set_motion(inputtino::PS5Joypad::ACCELERATION, motion.x, motion.y, motion.z);
+          break;
+        case LI_MOTION_TYPE_GYRO:
+          std::get<inputtino::PS5Joypad>(*gamepad->joypad).set_motion(inputtino::PS5Joypad::GYROSCOPE, motion.x, motion.y, motion.z);
+          break;
+      }
+    }
+  }
+
+  void
+  gamepad_battery(input_t &input, const gamepad_battery_t &battery) {
+    auto raw = (input_raw_t *) input.get();
+    auto gamepad = raw->gamepads[battery.id.globalIndex];
+    if (!gamepad) {
+      return;
+    }
+    // Only the PS5 controller supports motion
+    if (std::holds_alternative<inputtino::PS5Joypad>(*gamepad->joypad)) {
+      inputtino::PS5Joypad::BATTERY_STATE state;
+      switch (battery.state) {
+        case LI_BATTERY_STATE_CHARGING:
+          state = inputtino::PS5Joypad::BATTERY_CHARGHING;
+          break;
+        case LI_BATTERY_STATE_DISCHARGING:
+          state = inputtino::PS5Joypad::BATTERY_DISCHARGING;
+          break;
+        case LI_BATTERY_STATE_FULL:
+          state = inputtino::PS5Joypad::BATTERY_FULL;
+          break;
+        case LI_BATTERY_STATE_NOT_CHARGING:
+        case LI_BATTERY_PERCENTAGE_UNKNOWN:
+        case LI_BATTERY_STATE_UNKNOWN:
+        case LI_BATTERY_STATE_NOT_PRESENT:
+          state = inputtino::PS5Joypad::CHARGHING_ERROR;
+          break;
+      }
+      std::get<inputtino::PS5Joypad>(*gamepad->joypad).set_battery(state, battery.percentage / 2.55);  // TODO: 255 (0xFF) is 100%?
+    }
+  }
+
+  platform_caps::caps_t
+  get_capabilities() {
+    platform_caps::caps_t caps = 0;
+    // TODO: if has_uinput
+    caps |= platform_caps::pen_touch;
+
+    // We support controller touchpad input as long as we're not emulating X360
+    if (config::input.gamepad != "x360"sv) {
+      caps |= platform_caps::controller_touch;
+    }
+
+    return caps;
+  }
+
+  std::vector<std::string_view> &
+  supported_gamepads() {
+    static std::vector<std::string_view> gps {
+      "auto"sv, "x360"sv, "ds4"sv, "ps4"sv, "switch"sv
+    };
+
+    return gps;
+  }
+}  // namespace platf
```

Comment:
```suggestion
}  // namespace platf

```

Reply:
I think there's a config option to do it automatically on save, also to remove trailing whitespace.